[lld][LTO] Teach LTO to print pipeline passes

macurtis-amd opened the pull request:
I found this useful while debugging code generation differences between old and new offloading drivers.
No functional change (intended).

macurtis-amd committed:
[LTO] Teach LTO to print pipeline passes

macurtis-amd commented:
> It looks like from clang this is triggered by an internal LLVM option (`-print-pipeline-passes`). Since this is a debugging functionality, rather than add another LLD option can you just make the LTO code be guarded by the same option? Then it can just be triggered via `-Wl,-mllvm,-print-pipeline-passes`, and the advantage is that the internal option can then also be used from other linkers such as gold.

Thanks for the feedback. Latest revision reuses `-print-pipeline-passes` as you suggested.

teresajohnson commented on llvm/lib/LTO/LTOBackend.cpp:
```diff
@@ -335,6 +335,16 @@ static void runNewPMPasses(const Config &Conf, Module &Mod, TargetMachine *TM,
   if (!Conf.DisableVerify)
     MPM.addPass(VerifierPass());
 
+  if (PrintPipelinePasses) { // e.g. lld -mllvm=-print-pipeline-passes
```
No need for comment about lld

teresajohnson reviewed (commented):
Thanks - since this is now in LTO just make an LTO test using something like either llvm-lto or llvm-lto2 instead of lld. See examples in llvm/test/LTO/X86

macurtis-amd commented:
> Thanks - since this is now in LTO just make an LTO test using something like either llvm-lto or llvm-lto2 instead of lld. See examples in llvm/test/LTO/X86

Removed lld test and added LTO test. No more lld changes :)

teresajohnson reviewed (approved):
lgtm
